app/javascript/react/components/SessionsListView/SessionsListView.tsx

```diff
@@ -49,16 +89,33 @@ const SessionsListView: React.FC<SessionsListViewProps> = ({
     }
   };
 
-  return (
-    <S.SessionListViewStyle>
-      <S.SessionInfoTile>
-        <S.SessionListInfoContainer>
-          <S.SessionListTitle>{t("map.listSessions")}</S.SessionListTitle>
-          <S.SessionListTitle>
-            {t("map.results", { results: results })}
-          </S.SessionListTitle>
-        </S.SessionListInfoContainer>
+  const handleExportClick = () => {
+    if (NO_SESSIONS) {
+      setAlertMessage(t("exportDataModal.noResultsMessage"));
+      setShowAlert(true);
+    } else if (EXCEEDS_LIMIT) {
+      setAlertMessage(
+        t("exportDataModal.sessionLimitMessage", { limit: SESSIONS_LIMIT })
+      );
+      setShowAlert(true);
+    } else {
+      setShowExportPopup(true);
+    }
+  };
 
+  useAutoDismissAlert(showAlert, setShowAlert);
+
+  const ExportButtonOrComponent = () => {
```

Comment:
Hmm, wouldn't it be better to move this function outside of the session list view or at least outside of the main component to avoid recreating it on every render?